pkg/liqo-controller-manager/namespace-controller/namespace_controller.go

```diff
@@ -1,147 +1,82 @@
-limitations under the License.
-*/
-
 package namespacectrl
 
 import (
 	"context"
 
-	mapsv1alpha1 "github.com/liqotech/liqo/apis/virtualKubelet/v1alpha1"
-	liqoconst "github.com/liqotech/liqo/pkg/consts"
-
 	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/klog"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	ctrlutils "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	offv1alpha1 "github.com/liqotech/liqo/apis/offloading/v1alpha1"
+	liqoconst "github.com/liqotech/liqo/pkg/consts"
 )
 
-// NamespaceReconciler changes DesiredMapping field of NamespaceMaps if a remote Namespace must be created or deleted.
+// NamespaceReconciler covers the case in which the user adds the enabling liqo label to his namespace, and the
+// NamespaceOffloading resource associated with that namespace is created, if it is not already there.
 type NamespaceReconciler struct {
 	client.Client
 	Scheme *runtime.Scheme
 }
 
 const (
-	mappingLabel                 = "mapping.liqo.io"
-	offloadingLabel              = "offloading.liqo.io"
-	offloadingPrefixLabel        = "offloading.liqo.io/"
-	namespaceControllerFinalizer = "namespace-controller.liqo.io/finalizer"
+	annotationKey   = "resource-controlled-by"
+	annotationValue = "This resource is created by the Namespace Controller"
 )
 
-// Reconcile checks presence of right labels on the Namespace.
+// Reconcile covers the case in which the user adds the enabling liqo label to his namespace, and the
+// NamespaceOffloading resource associated with that namespace is created, if it is not already there.
 func (r *NamespaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	namespace := &corev1.Namespace{}
-	if err := r.Get(context.TODO(), req.NamespacedName, namespace); err != nil {
+	if err := r.Get(ctx, req.NamespacedName, namespace); err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Infof("There is no namespace called '%s' in the cluster", req.Name)
+			return ctrl.Result{}, nil
+		}
 		klog.Errorf("%s --> Unable to get namespace '%s'", err, req.Name)
-		return ctrl.Result{}, client.IgnoreNotFound(err)
-	}
-
-	namespaceMaps := &mapsv1alpha1.NamespaceMapList{}
-	if err := r.List(context.TODO(), namespaceMaps); err != nil {
-		klog.Error(err, " --> Unable to List NamespaceMaps")
 		return ctrl.Result{}, err
 	}
 
-	if len(namespaceMaps.Items) == 0 {
-		klog.Info(" No namespaceMaps at the moment in the cluster")
-		return ctrl.Result{}, nil
+	// Check if a NamespaceOffloading Resource called "offloading" is present in this Namespace.
+	namespaceOffloading := &offv1alpha1.NamespaceOffloading{}
+	var namespaceOffloadingIsPresent bool
+	checkPresenceErr := r.Get(ctx, types.NamespacedName{
+		Namespace: namespace.Name,
+		Name:      liqoconst.DefaultNameNamespaceOffloading,
```

Comment:
```suggestion
		Name:      liqoconst.DefaultNameNamespaceOffloadingName,
```

For the sake of clarity

Reply:
Do you mean `DefaultNamespaceOffloadingName` instead of `DefaultNameNamespaceOffloading` @giorio94 ?

Reply:
Yes, sorry, I missed there was already the `Name` word. I feel it clearer at the end

Reply:
thank you